src/main/java/build/buildfarm/worker/Executor.java

```diff
@@ -212,7 +217,7 @@ private long executePolled(
               "", // executingMetadata.getStderrStreamName(),
               resultBuilder);
 
-      if (resource.isReferenced()) {
+      if (resource.isReferenced() && errorOperationRemainingResources) {
```

Comment:
Swap these checks. The method may be a more substantial performance impact than the config check.

Reply:
fix